ppmpy/ppm.py

```diff
@@ -11772,6 +11782,251 @@ def get_spherical_coordinates(self, theta, phi):
+        '''
+        iter_list = [int(i) for i in range(num_rays)]
+        i=0
+        rays=[]
+        while i<data_points: #i is the number of points in our radii array
+            rays.append(data[i][ind]) # gives us the data from our specific directions
+            i+=1
+        rays = np.array(rays)
+        
+        rays_data = dict()
+        for ray_id in iter_list:
+            raydir = np.array([rays[i][ray_id] for i,_ in enumerate(rays)])
+            rays_data[ray_id] = raydir
+        return [rays_data]
+        
+    def ray_analysis(self, varloc, fname, xlim = [0,2685] ,num_rays = 96, logy = False, singleplot = False, ifig = 1): 
+        '''
+        Takes one of 3 variables and creates a radial ray analysis plot in comparison to the spherical average.
+        
+        Parameters
+        ----------
+        varloc: str
+            One of 3: '|ut|' , '|w|' or 'fv'
+        fname: int
+            The dump number
+        xlim: list
+            The xlim that you would like to see for your plot
+        num_rays: even integer
+            The number of rays you would like analyzed. A factor of 4 if you would like to see the 4 panel plot. 
+        logy:
+            Would you like the log of the y value (useful for FV mostly)
+        singleplot:
+            If you would like only one plot, otherwise defaults to 4 subplots
+        ifig:
+            The figure number defaulted to 1
+        '''
+        
+        if num_rays%4 != 0 and singleplot == False:
+            return 'Error: Please make the number of rays you would like analyzed a factor of 4, or change singleplot = True.'
+        
+        # needed variables for plotting
+        iter_list = [int(i) for i in range(num_rays)]
+        needed_data = self.__data_setup(varloc, fname, num_rays)
+        data_rays = self.__data_sort(needed_data[1], needed_data[2], needed_data[3], num_rays)
+        radii = needed_data[4]
+        run_name = self._run_id
+        
+        # indexes for the iteration of the number of rays...
+        ind1 = int(num_rays/4 -1)
+        ind2 = int(num_rays/4)
+        ind3 = int(ind2*2 - 1)
+        ind4 = int(ind3 + 1)
+        ind5 = int(ind2 * 3 - 1)
+        ind6 = int(ind5 + 1)
+        ind7 = int(num_rays - 1)
+        
+        # this allows for proper linestyle and colour cycling of the plots
+        linecycle = cycle(lll) 
+        colour_cycle = cycle(CB_color_cycle)
+        
+        if singleplot == True:
+            pl.close(ifig); pl.figure(ifig, figsize=(12.5,6))
+            for i in iter_list:
+                pl.plot(radii, data_rays[0][i], next(linecycle), c = next(colour_cycle), linewidth=0.5)
+            pl.plot(radii,needed_data[0] ,'-r',markevery=25, label = 'Spherically Averaged')
+            pl.title('{} - {} : Dump {}'.format(run_name,varloc,fname));pl.xlim(xlim);pl.legend()
+            
+            if varloc=='Ut':
```

Comment:
I need to change this, it appears twice more but the 'Ut' varloc name should be '|ut|' instead.